### PR TITLE
Addition of Vax Stats

### PR DESCRIPTION
**Pull request by handyjellyfish**
Adds a Vax Stats feature counting down to certain whole number percentages. Can be run in conjunction with (default), or separately from the existing code output. 

Recommend running it with AUS,NSW,VIC for good coverage and info.

Let me know thoughts!

image

### CODE REVIEW

File: slack_bot.py [hunk 1]
```diff
@@ -37,14 +29,113 @@ def post_messages_to_slack(self, messages: List[str]) -> SlackResponse:
...
+        return self.post_messages_to_slack(blocks)
+
+    def execute_for_vax_stats(self, vax_data: Dict) -> SlackResponse:
+        # Generate the messages for each state/country code
+        blocks = [
+            {
+                "type": "header",
+                "text": {
+                    "type": "plain_text",
+                    "text": ":dart: Vax Stats"
```
Comment (B0yc3y) [inline slack_bot.py:58]: I'm not too sure if this is markdown or plain text. It's the preview notification text

File: slack_bot.py [hunk 2]
```diff
@@ -37,14 +29,113 @@ def post_messages_to_slack(self, messages: List[str]) -> SlackResponse:
...
+                }
+            },
+            {
+                "type": "divider"
+            }
+        ]
+        
+        for code in vax_data:
+            blocks.extend(self.generate_vax_stats_for_code(vax_data[code]))
```
Comment (B0yc3y) [inline slack_bot.py]:
```suggestion
        ].extend( [self.generate_vax_stats_for_code(vax_data[code]) for code in vax_data] )
```
Nitpicky, but this is a little more pythonic :)

**Review by B0yc3y — APPROVED** [review]
Thanks agian Rich, Two very minor things. Have you been able to test this anywhere or do I need to put it through its paces a little?

**Comment by handyjellyfish** [comment]
> Thanks agian Rich, Two very minor things. Have you been able to test this anywhere or do I need to put it through its paces a little?

I have my own test space that I've been running it on. And it's running with current data. Seems pretty stable. Light on tests with the docker container, but have tested it working. Added a test for the display data as well.

One minor thing is that I'm running with individual rolling averages for first and second doses, whereas covidlive has switched to an avg+6week calc so the numbers don't match entirely. No massive issue, they are still predictions based on current data and will slowly merge to being accurate as time passes. Might look to see if I can figure out the maths for the 6week calc, or feel free to take it on too :D

File: slack_bot.py (hunk 1 above)
Comment (handyjellyfish) [reply slack_bot.py:58]: https://api.slack.com/reference/block-kit/blocks#header header is always plain_text - can't be mrkdwn - assume that's what you're querying? :)

**Comment by B0yc3y** [comment]
Great, sounds good! 

> Might look to see if I can figure out the maths for the 6week calc, or feel free to take it on too :D
Email the covidlive maintainer. contact@covidlive.com.au he's pretty snappy with replies, and might be able to just share their logic.

**Pull request merged**